src/Embeddings/VectorStores/Redis/RedisVectorStore.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace LLPhant\Embeddings\VectorStores\Redis;
+
+use LLPhant\Embeddings\Document;
+use LLPhant\Embeddings\DocumentUtils;
+use LLPhant\Embeddings\VectorStores\VectorStoreBase;
+use Predis;
+use Predis\Command\Argument\Search\CreateArguments;
+use Predis\Command\Argument\Search\DropArguments;
+use Predis\Command\Argument\Search\SchemaFields\TextField;
+use Predis\Command\Argument\Search\SchemaFields\VectorField;
+use Predis\Command\Argument\Search\SearchArguments;
+
+class RedisVectorStore extends VectorStoreBase
+{
+    public const LLPHANT_INDEX = 'llphant';
+
+    public Predis\Client $client;
+
+    public function __construct(Predis\Client $client)
+    {
+        $this->client = $client;
+
+        try {
+            // Drop index and delete all data in it
+            $client->ftdropindex(self::LLPHANT_INDEX, (new DropArguments())->dd());
+        } catch (Predis\Response\ServerException $e) {
+            if ($e->getMessage() !== 'Unknown Index name') {
+                throw $e;
+            }
+        }
+    }
+
+    public function addDocument(Document $document): void
+    {
+        $this->client->jsonmset([...$this->generateRedisJsonSetArguments($document)]);
+    }
+
+    /**
+     * @param  Document[]  $documents
+     */
+    public function addDocuments(array $documents): void
+    {
+        if ($documents === []) {
+            return;
+        }
+
+        $redisArgs = [];
+        foreach ($documents as $document) {
+            array_push($redisArgs, ...$this->generateRedisJsonSetArguments($document));
+        }
+
+        $this->client->jsonmset(...$redisArgs);
+    }
+
+    /**
+     * Return docs most similar to the embedding.
+     *
+     * @param  float[]  $embedding
+     * @param  array<string, string|int>|array<mixed[]>  $additionalArguments
+     * @return mixed[]
+     */
+    public function similaritySearch(array $embedding, int $k = 4, array $additionalArguments = []): array
+    {
+        $vectorDimension = count($embedding);
+        $this->createIndexIfMissing($vectorDimension, $k);
```

Comment:
it should raise an error if it is not created or return empty array

Reply:
`createIndex` peut déjà throw lors du `ftcreate`
![Capture d’écran 2023-10-05 à 10 09 03](https://github.com/theodo-group/LLPhant/assets/55784802/27a7f3c4-70d8-4966-b0db-fa91daa9a206)
Tu veux throw une erreur custom en mettant createIndexIfMissing dans un try catch ?